app/content/profile/index.js

```diff
@@ -65,6 +85,157 @@ function Index(props) {
     isLoading: drillInfoIsLoading,
   } = useDrillInfo();
 
+  // ref
+  const bottomSheetModalRef = useRef(null);
+
+  // variables
+  const initialSnapPoints = useMemo(() => [355, 455, 730], []);
+  const expandedSnapPoints = useMemo(() => [460, 570, 860], []); // Adjusted snap points for expanded content
+  const [snapPoints, setSnapPoints] = useState(initialSnapPoints);
+
+  // callbacks
+  const handlePresentModalPress = useCallback(() => {
+    bottomSheetModalRef.current?.present();
+  }, []);
+  const handleSheetChanges = useCallback((index) => {
+    // console.log("handleSheetChanges", index);
+  }, []);
+
+  const [currentName, setCurrentName] = useState("");
+  const [newName, setNewName] = useState("");
+  const [email, setEmail] = useState("");
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [passwordInputVisible, setPasswordInputVisible] = useState(false);
+
+  const [snackbarVisible, setSnackbarVisible] = useState(false); // State to toggle snackbar visibility
+  const [snackbarMessage, setSnackbarMessage] = useState(""); // State to set snackbar message
+
+  const [dialogVisible, setDialogVisible] = useState(false);
+  const [dialogTitle, setDialogTitle] = useState("");
+  const [dialogMessage, setDialogMessage] = useState("");
+
+  const handleSnackbarDismiss = () => {
+    setSnackbarVisible(false); // Dismiss snackbar
+  };
+
+  useEffect(() => {
```

Comment:
O god pls no, just use a loading screen and wait for userData to populate. Don't do this useEffect train...